src/PrestaShopBundle/Translation/Provider/TranslationsCatalogueProvider.php

```diff
@@ -0,0 +1,263 @@
+            $userTranslatedCatalogue,
+            $treeDomain,
+            $search
+        );
+    }
+
+    /**
+     * @param TypeInterface $providerType
+     * @param string $locale
+     * @param array $search
+     *
+     * @return array
+     *
+     * @throws \PrestaShopBundle\Exception\NotImplementedException
+     */
+    public function getCatalogue(
+        TypeInterface $providerType,
+        string $locale,
+        array $search = []
+    ): array {
+        $provider = $this->providerFactory->build($providerType);
+
+        $defaultCatalogue = $provider->getDefaultCatalogue($locale);
+        $defaultCatalogueMessages = $defaultCatalogue->all();
+        if (empty($defaultCatalogueMessages)) {
+            return [];
+        }
+        $fileTranslatedCatalogue = $provider->getFileTranslatedCatalogue($locale);
+        $userTranslatedCatalogue = $provider->getUserTranslatedCatalogue($locale);
+
+        $translations = [];
+
+        foreach ($defaultCatalogueMessages as $domain => $messages) {
+            $missingTranslations = 0;
+            $translations[$domain] = [];
+
+            foreach ($messages as $translationKey => $translationValue) {
+                $data = [
+                    'default' => $translationKey,
+                    'xliff' => $fileTranslatedCatalogue->defines($translationKey, $domain)
+                        ? $fileTranslatedCatalogue->get($translationKey, $domain)
+                        : null,
+                    'database' => $userTranslatedCatalogue->defines($translationKey, $domain)
+                        ? $userTranslatedCatalogue->get($translationKey, $domain)
+                        : null,
+                ];
+
+                // if search is empty or is in catalog default|xliff|database
+                if (empty($search) || $this->dataContainsSearchWord($data, $search)) {
+                    $translations[$domain][$translationKey] = $data;
+
+                    if (null === $data['xliff'] && null === $data['database']) {
+                        ++$missingTranslations;
+                    }
+                }
+            }
+
+            $translations[$domain]['__metadata'] = [
+                'count' => count($translations[$domain]),
+                'missing_translations' => $missingTranslations,
+            ];
+        }
+
+        unset($catalogues);
+
+        ksort($translations);
+
+        return $translations;
+    }
+
+    /**
+     * Filters the catalogue so that only domains matching the filters are kept
+     *
+     * @param string $locale
+     * @param MessageCatalogueInterface $catalogue
+     * @param string $domain
+     *
+     * @return MessageCatalogueInterface
+     */
+    private function filterCatalogue(MessageCatalogueInterface $catalogue, string $locale, string $domain): MessageCatalogueInterface
+    {
+        $filteredCatalogue = [];
+        $filenameFilters = ['#^' . preg_quote($domain, '#') . '([A-Za-z]|\.|$)#'];
```

Comment:
The array is not needed here, and I'm afraid with the regex (even if it came from the legacy. this kind of regex match
```
ModulesBannerShopA
ModulesBannerShopBaiozjdpoajzdojazodjaozjdoajzd
ModulesBannerShop.Something
ModulesBannerShop
```